Include/datetime.h

```diff
@@ -187,17 +187,35 @@ typedef struct {
 
 #define PyDateTime_CAPSULE_NAME "datetime.datetime_CAPI"
 
+/* Internal use */
+typedef struct {
+    PyDateTime_CAPI api;
+    PyDateTime_CAPI *(*_get_capi_by_interp)(void);
+    void (*_set_capi_by_interp)(PyDateTime_CAPI *);
+} _pydatetime_capi;
 
 /* This block is only used as part of the public API and should not be
  * included in _datetimemodule.c, which does not use the C API capsule.
  * See bpo-35081 for more details.
  * */
 #ifndef _PY_DATETIME_IMPL
 /* Define global variable for the C API and a macro for setting it. */
-static PyDateTime_CAPI *PyDateTimeAPI = NULL;
-
-#define PyDateTime_IMPORT \
-    PyDateTimeAPI = (PyDateTime_CAPI *)PyCapsule_Import(PyDateTime_CAPSULE_NAME, 0)
+static PyDateTime_CAPI *
+_get_pydatetime_capi_dummy(void)
+{
+    return NULL;
+}
+static PyDateTime_CAPI *(*_get_pydatetime_capi)(void) = _get_pydatetime_capi_dummy;
+#define PyDateTimeAPI _get_pydatetime_capi()
+
+static inline void pydatetime_import(void) {
```

Comment:
I would prefer to not leak such implementation detail at the ABI level, and so use a regular function rather than a static inline function. I would prefer that _pydatetime_capi is private.

Reply:
How can I resolve non-Windows errors?  I'm not sure how to use private struct members here.

```
Error:  _testcapi failed to import: /home/runner/work/cpython/cpython-builddir/
build/lib.linux-x86_64-3.13-pydebug/_testcapi.cpython-313d-x86_64-linux-gnu.so:
undefined symbol: _PyDateTimeAPI_Get
```